File: Krishna/Arrays/Rotate Image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) 
    {
         int temp=0;
        for(int i=0;i<matrix.size();i++){
            for(int j=i+1;j<matrix.size();j++){
               swap(matrix[i][j],matrix[j][i]);
            }
        }
        int a=0,b=matrix.size()-1;
        
        while(a!=b && a<b){
             for(int i=0;i<matrix.size();i++)
                 swap(matrix[i][a],matrix[i][b]);
            a++;
            b--;
                 }
    }
};

File: Rohit/RotateImage.java
class Solution {
    public void rotate(int[][] matrix) {
        int rows=matrix.length;
        int cols=matrix[0].length;
        
        for(int i=0;i<rows;i++)
        {
            for(int j=i;j<rows;j++)
            {
                int temp=matrix[i][j];         //transpose 
                matrix[i][j]=matrix[j][i];
                matrix[j][i]=temp;
            }
        }
        
        
        for(int i=0;i<rows;i++)
        {
            for(int j=0;j<rows/2;j++)
            {
                int temp=matrix[i][j];
                matrix[i][j]=matrix[i][matrix.length-1-j];   //reverse
                matrix[i][matrix.length-1-j]=temp;
            }
        }
        
        
    }
}
